packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -241,6 +241,40 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
     }());
   }
 
+  @override
+  void insert(RenderBox child, { RenderBox after }) {
+    assert(!_keepAliveBucket.containsValue(child));
+    super.insert(child, after: after);
+    assert(firstChild != null);
+    if (integrityEnabled){
+      verifyIntegrity();
+    }
+  }
+
+  /// Move child to a different location in the list.
+  /// There are two scenarios:
+  /// 1, the child is in _keepAliveBucket
+  /// In this case, the child is no longer in the rendering tree. we just
+  /// need to take it out from bucket, readopt it, and insert back to list
+  /// 2, the child is in rendering tree
+  /// we can call the super.move and update parentData to new slot
+  @override
+  void move(RenderBox child, { RenderBox after }) {
+    final SliverMultiBoxAdaptorParentData childParentData = child.parentData;
+    if (_keepAliveBucket.containsKey(childParentData.index)) {
+      _keepAliveBucket.remove(childParentData.index);
```

Comment:
For future readers of this code it would probably be helpful to add a comment explaining why we don't have to add the kept alive child back to the keep alive bucket.

Reply:
done